content/lang/tr/guide/forms.md

```diff
@@ -6,27 +6,23 @@ permalink: '/guide/forms'
 
-React's documentation on ["Controlled" Components](https://facebook.github.io/react/docs/forms.html#controlled-components) and ["Uncontrolled" Components](https://facebook.github.io/react/docs/forms.html#uncontrolled-components) is immensely useful in understanding how to take HTML forms, which have bidirectional data flow, and make use of them from the context of a Component-based Virtual DOM renderer, which generally has unidirectional data flow.
+Genellikle, her zaman _Kontrollü_ bileşenleri kullanmalısınız.Ancak bağımsız Bileşenler oluştururken veya 3.parti UI kütüphaneleri kullanırken, non-preact fonksiyonelliği için kendi bileşeninizi bir bağlama noktası olarak kullanmak daha kullanışlı ve basit olabilir.
 
-Generally, you should try to use _Controlled_ Components at all times.  However, when building standalone Components or wrapping third-party UI libraries, it can still be useful to simply use your component as a mount point for non-preact functionality.  In these cases, "Uncontrolled" Components are nicely suited to the task.
-
-
-## Checkboxes & Radio Buttons
-
-Checkboxes and radio buttons (`<input type="checkbox|radio">`) can initially cause confusion when building controlled forms. This is because in an uncontrolled environment, we would typically allow the browser to "toggle" or "check" a checkbox or radio button for us, listening for a change event and reacting to the new value.  However, this technique does not transition well into a world view where the UI is always updated automatically in response to state and prop changes.
-
-> **Walk-Through:** Say we listen for a "change" event on a checkbox, which is fired when the checkbox is checked or unchecked by the user.  In our change event handler, we set a value in `state` to the new value received from the checkbox.  Doing so will trigger a re-render of our component, which will re-assign the value of the checkbox to the value from state.  This is unnecessary, because we just asked the DOM for a value but then told it to render again with whatever value we wanted.
+## Checkboxe'lar & Radio Button'lar
 
+Checkbox'lar ve radio button'lar (`<input type="checkbox|radio">`) kontrollü formlar oluştururken başlangıçta karışık gelebilir.Bunun sebebi kontrolsüz bir ortamda, tarayıcıya bir checkbox veya radio buton'u bizim için "değiştir" veya "kontrol et", gibi değişiklik olayını bizim için dinlemesi ve yeni değere tepki vermesi için izin veririz.Ancak, bu teknik kullanıcı arayüzünün state ve props değişikliklerine yanıt olarak her zaman otomatik olarak güncelleyeceği anlamına gelmez.
```

Comment:
Checkbox'lar ve radio button'lar (`<input type="checkbox|radio">`) kontrollü formlar oluştururken karışık gelebilir. Bunun sebebi kontrolsüz bir ortamda, tarayıcıya bir checkbox veya radio buton'u bizim için "toggle" veya "check" gibi bir change event'ı bizim için dinlemesi ve yeni değere tepki vermesi için izin veririz. Ancak, bu teknik kullanıcı arayüzünün state ve prop değişikliklerine yanıt olarak her zaman otomatik olarak güncelleyeceği anlamına gelmez.